app/src/main/java/woowacourse/payments/model/CardInfo.kt

```diff
@@ -1,30 +1,80 @@
 package woowacourse.payments.model
 
 import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 
 data class CardInfo(
     val cardNumber: String,
     val expireDate: LocalDate,
     val ownerName: String,
-    val password: String
+    val password: String,
 ) {
+    init {
+        require(cardNumber.length == CARD_NUMBER_MAX_SIZE) {
+            "카드 번호는 $CARD_NUMBER_MAX_SIZE 자리여야 합니다."
+        }
+        require(ownerName.length <= OWNER_NAME_MAX_SIZE) {
+            "카드 소유자 이름은 $OWNER_NAME_MAX_SIZE 자리 이하여야 합니다."
+        }
+        require(password.length == PASSWORD_MAX_SIZE) {
+            "비밀번호는 $PASSWORD_MAX_SIZE 자리여야 합니다."
+        }
+    }
+
     companion object {
-        fun checkIsValidMonth(expireDate: String): Boolean = expireDate
+        fun createOrNull(
+            cardNumber: String,
+            expireDate: String,
+            ownerName: String,
+            password: String,
+        ): CardInfo? {
+            val formattedCardNumber = (cardNumber)
+            val formattedExpireDateString = formatExpireDate(expireDate)
+            val formattedOwnerName = formatOwnerName(ownerName)
+            val formattedPassword = formatPassword(password)
+            val parsedExpireDate =
+                runCatching {
+                    LocalDate.parse(
+                        "01$formattedExpireDateString",
+                        DateTimeFormatter.ofPattern("ddMMyy"),
+                    )
+                }.getOrNull()
+
+            if (formattedCardNumber.length != CARD_NUMBER_MAX_SIZE) return null
+            if (formattedExpireDateString.length != EXPIRE_DATE_MAX_SIZE) return null
+            if (formattedOwnerName.length > OWNER_NAME_MAX_SIZE) return null
+            if (formattedPassword.length != PASSWORD_MAX_SIZE) return null
+            if (parsedExpireDate == null) return null
```

Comment:
```suggestion
            when {
                formattedCardNumber.length != CARD_NUMBER_MAX_SIZE -> return null
                formattedExpireDateString.length != EXPIRE_DATE_MAX_SIZE -> return null
                formattedOwnerName.length > OWNER_NAME_MAX_SIZE -> return null
                formattedPassword.length != PASSWORD_MAX_SIZE -> return null
                parsedExpireDate == null -> return null
            }
```

어느쪽이 가독성이 더 좋아보이나요? 이건 밀러의 재량입니다 👀 코틀린의 장점을 항상 활용해보세요 :)

Reply:
반영했습니다 ~ 
[refactor: 복잡한 if절을 when문으로 변경](https://github.com/woowacourse/android-payments/pull/44/commits/536e7efc462464d17051729ff4c3257fc5d105ff)